load.php

```diff
@@ -507,171 +262,44 @@ function perflab_maybe_remove_object_cache_dropin() {
-			 */
-			if ( empty( $active_modules_with_plugins ) ) {
-				return;
-			}
-
-			$current_user = wp_get_current_user();
-
-			/*
-			 * Disable WordPress pointers for specific users based on conditions.
-			 *
-			 * Checks if there is a large user count on the site. If true,
-			 * disables pointers for the current user only. Otherwise, disables
-			 * pointers for users with the same role as the current user.
-			 */
-			if ( wp_is_large_user_count() ) {
-				perflab_undismiss_module_migration_pointer( $current_user );
-			} else {
-				$current_user_roles = $current_user->roles;
-				$current_user_role  = array_shift( $current_user_roles );
-
-				$args = array(
-					'role'       => $current_user_role,
-					'meta_query' => array(
-						array(
-							'key'     => 'dismissed_wp_pointers',
-							'value'   => 'perflab-module-migration-pointer',
-							'compare' => 'LIKE',
-						),
-					),
-				);
-
-				$users = get_users( $args );
-
-				foreach ( $users as $user ) {
-					perflab_undismiss_module_migration_pointer( $user );
-				}
-			}
-		}
-	}
-
-	// Get the list of modules that were deactivated, and load the deactivate.php files if they exist.
-	if ( ! empty( $old_value ) ) {
-		foreach ( $old_value as $module => $module_settings ) {
-			if ( ! empty( $module_settings['enabled'] ) && ( empty( $value[ $module ] ) || empty( $value[ $module ]['enabled'] ) ) ) {
-				perflab_deactivate_module( PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module );
-			}
-		}
-	}
-
-	return $value;
-}
-
 /**
- * Reverts the module migration pointer dismissal for the given user.
+ * Redirects module pages to the performance feature page.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  *
- * @param WP_User $user The WP_User object.
+ * @global $plugin_page
  */
-function perflab_undismiss_module_migration_pointer( $user ) {
-	$dismissed = array_filter( explode( ',', (string) get_user_meta( $user->ID, 'dismissed_wp_pointers', true ) ) );
+function perflab_no_access_redirect_module_to_performance_feature_page() {
+	global $plugin_page;
 
-	$pointer_index = array_search( 'perflab-module-migration-pointer', $dismissed, true );
-	if ( false === $pointer_index ) {
+	if ( 'perflab-modules' !== $plugin_page ) {
 		return;
 	}
 
-	unset( $dismissed[ $pointer_index ] );
-	$dismissed = implode( ',', $dismissed );
-
-	update_user_meta( $user->ID, 'dismissed_wp_pointers', $dismissed );
-}
-
-/**
- * Activate a module.
- *
- * Runs the activate.php file if it exists.
- *
- * @since 1.8.0
- *
- * @param string $module_dir_path The module's directory path.
- */
-function perflab_activate_module( $module_dir_path ) {
-	$module_activation_file = $module_dir_path . '/activate.php';
-	if ( ! file_exists( $module_activation_file ) ) {
-		return;
-	}
-	$module = require $module_activation_file;
-	if ( ! is_callable( $module ) ) {
-		return;
+	if ( current_user_can( 'manage_options' ) ) {
+		wp_safe_redirect(
+			add_query_arg( 'page', PERFLAB_SCREEN )
+		);
+		exit;
 	}
```

Comment:
This would be better (as in more testable) if the `exit` only happens conditionally based on whether the redirect happens:

```suggestion
	if ( 
		current_user_can( 'manage_options' ) && 
		wp_safe_redirect( add_query_arg( 'page', PERFLAB_SCREEN ) ) 
	) {
		exit;
	}
```

But apparently this isn't being tested right now, so not critical to change here.